### PR TITLE
Move create release from models to a standalone package

### PR DESCRIPTION
**Pull request by lunny**

### CODE REVIEW

File: routers/api/v1/repo/release.go [hunk 1]
```diff
@@ -8,8 +8,8 @@ import (
 	"code.gitea.io/gitea/models"
 	"code.gitea.io/gitea/modules/context"
 	"code.gitea.io/gitea/modules/setting"
-
 	api "code.gitea.io/gitea/modules/structs"
+	release_service "code.gitea.io/gitea/services/release"
```
Comment (mrsdizzie) [inline routers/api/v1/repo/release.go]: The underscore seems non-go like. Could this not just be "release" ?

Comment (lunny) [reply routers/api/v1/repo/release.go]: If all variable named `rel` but not `release` that I think use `release` OK, otherwise it will conflict.

Comment (lunny) [reply routers/api/v1/repo/release.go]: I renamed it to releaseservice

**Review by guillep2k — APPROVED** [review]

File: services/release/release.go [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+}
+
+// UpdateRelease updates information of a release.
+func UpdateRelease(doer *models.User, gitRepo *git.Repository, rel *models.Release, attachmentUUIDs []string) (err error) {
+	if err = createTag(gitRepo, rel); err != nil {
+		return err
+	}
+	rel.LowerTagName = strings.ToLower(rel.TagName)
+
+	err = models.UpdateRelease(rel)
```
Comment (lafriks) [inline services/release/release.go]:
```suggestion
	if err = models.UpdateRelease(rel); err != nil {
```

File: services/release/release.go [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+	if err != nil {
+		return err
+	}
+
+	err = rel.LoadAttributes()
```
Comment (lafriks) [inline services/release/release.go]: same here

File: services/release/release.go [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+	} else if isExist {
+		return models.ErrReleaseAlreadyExist{
+			TagName: rel.TagName,
+		}
+	}
+
+	if err = createTag(gitRepo, rel); err != nil {
+		return err
+	}
+	rel.LowerTagName = strings.ToLower(rel.TagName)
```
Comment (lafriks) [inline services/release/release.go]: same here

File: services/release/release.go [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+
+	err = models.InsertRelease(rel)
```
Comment (lafriks) [inline services/release/release.go]: same here

File: services/release/release.go [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+	if err != nil {
+		return err
+	}
+
+	err = models.AddReleaseAttachments(rel.ID, attachmentUUIDs)
```
Comment (lafriks) [inline services/release/release.go]: and here :)

**Review by lafriks — CHANGES_REQUESTED** [review]

**Comment by lunny** [comment]
@lafriks done.

**Review by lafriks — APPROVED** [review]

**Review by guillep2k — APPROVED** [review]

**Pull request merged**